packages/flutter/lib/src/widgets/image.dart

```diff
@@ -1119,8 +1121,9 @@ class _ImageState extends State<Image> with WidgetsBindingObserver {
     super.didUpdateWidget(oldWidget);
     if (_isListeningToStream &&
         (widget.loadingBuilder == null) != (oldWidget.loadingBuilder == null)) {
-      _imageStream!.removeListener(_getListener());
+      final ImageStreamListener oldListener = _getListener();
       _imageStream!.addListener(_getListener(recreateListener: true));
+      _imageStream!.removeListener(oldListener);
```

Comment:
This dance is necessary because otherwise the removal of the oldListener may dispose the image?

Reply:
Correct.